pytensor/compile/builders.py

```diff
@@ -401,6 +401,19 @@ def __init__(
             inline.
         """
 
+        if len(inputs) != len(set(inputs)):
+            var_counts = {var: inputs.count(var) for var in inputs}
+            duplicated_inputs = [
+                var.name if var.name is not None else str(var)
```

Comment:
The string representation of care already does this?
```suggestion
                var
```